src/neurostatslib/solver.py

```diff
@@ -0,0 +1,593 @@
+        """
+        if mask.ndim != 2:
+            raise ValueError(
+                "`mask` must be 2-dimensional. "
+                f"{mask.ndim} dimensional mask provided instead!"
+            )
+
+        if mask.shape[0] == 0:
+            raise ValueError(f"Empty mask provided! Mask has shape {mask.shape}.")
+
+        if jnp.any((mask != 1) & (mask != 0)):
+            raise ValueError("Mask elements be 0s and 1s!")
+
+        if mask.sum() == 0:
+            raise ValueError("Empty mask provided!")
+
+        if jnp.any(mask.sum(axis=0) > 1):
+            raise ValueError(
+                "Incorrect group assignment. Some of the features are assigned "
+                "to more then one group."
+            )
+
+        if not jnp.issubdtype(mask.dtype, jnp.floating):
+            raise ValueError(
+                "Mask should be a floating point jnp.ndarray. "
+                f"Data type {mask.dtype} provided instead!"
+            )
+
+    @abc.abstractmethod
+    def get_prox_operator(
+        self,
+    ) -> Callable[
+        [Tuple[jnp.ndarray, jnp.ndarray], float, float], Tuple[jnp.ndarray, jnp.ndarray]
+    ]:
+        """
+        Abstract method to retrieve the proximal operator for this solver.
+
+        Returns
+        -------
+        :
+            The proximal operator, which typically applies a form of regularization.
+        """
+        pass
+
+    def instantiate_solver(
+        self,
+        loss: Callable[
+            [Tuple[jnp.ndarray, jnp.ndarray], jnp.ndarray, jnp.ndarray], jnp.ndarray
+        ],
+    ) -> Callable[
+        [Tuple[jnp.ndarray, jnp.ndarray], jnp.ndarray, jnp.ndarray], jaxopt.OptStep
+    ]:
+        """
+        Instantiate the solver with the provided loss function and proximal operator.
+
+        Parameters
+        ----------
+        loss :
+            The original loss function to be optimized.
+
+        Returns
+        -------
+        :
+            A function that runs the solver with the provided loss and proximal operator.
+        """
+        self._check_is_callable_from_jax(loss)
+
+        solver_kwargs = self.solver_kwargs.copy()
+        solver_kwargs["fun"] = loss
+        solver_kwargs["prox"] = self.get_prox_operator()
+
+        run_kwargs = dict(hyperparams_prox=self.regularizer_strength)
+
+        return self.get_runner(solver_kwargs, run_kwargs)
+
+
+class LassoSolver(ProxGradientSolver):
+    """
+    Solver for optimization using the Lasso (L1 regularization) method with Proximal Gradient.
+
+    This class is a specialized version of the ProxGradientSolver with the proximal operator
+    set for L1 regularization (Lasso). It utilizes the `jaxopt` library's proximal gradient optimizer.
+    """
+
+    def __init__(
+        self,
+        solver_name: str = "ProximalGradient",
+        solver_kwargs: Optional[dict] = None,
+        regularizer_strength: float = 1.0,
+        mask: Optional[Union[NDArray, jnp.ndarray]] = None,
```

Comment:
should the regular lasso have mask arg? I thought only the group one used it

Reply:
and the fact that it does nothing is confusing

Reply:
it is confusing and i got rid of it